test/test/nativemem/profile_with_dlopen.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "asprof.h"
+#include <dlfcn.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define ASSERT_NO_DLERROR()  \
```

Comment:
What's the advantage of `#define` over a function?

Reply:
We can exit directly, without checking the return value of a new function. I used macros here to reduce repetition.